### DISCUSSION TITLE
Change data tpe for graph

### DISCUSSION THREAD

**Discussion by AndiLeni** [sqlitebrowser/sqlitebrowser#2853] [Q&A]
Hello,
is it possible to change these data-types?
![grafik](https://user-images.githubusercontent.com/13385771/135716072-49681c72-4c87-4f92-b620-d96c4359695a.png)

My columns have datetime objects, but DB Browser classifies them as a time value and hence displays the graph wrong.

**Comment by chrisjlocke** [comment]
SQLite doesn't have 'datetime' data types - only text, integers, real and blobs.
In the table structure, how is the 'datetime' column specified - text or a custom type of 'datetime' ?

**Reply by AndiLeni** [reply]
It is assigned as "text".

**Comment by chrisjlocke** [comment]
Aah, just saw DB4S recognises dates and sets the type accordingly.

![image](https://user-images.githubusercontent.com/7872218/135720537-26d257fc-d2f3-45d0-acc0-578e929fdb0a.png)

The datetime isn't selected in your plot - what does it show if you do?
When you say the graph is 'wrong', what is wrong about it?

**Reply by AndiLeni** [reply]
The main problem is, I am not able to change the type. 
F.e. some entries in my DB contain float vales, but some cells are empty, because there is no data available. 
DB Browser then automatically selects these columns as "label", but I want and need them to be parsed as floats.
Is there somewhere an option where the type for the diagram can be selected manually?

I was able to correctly display datetimes by changing them to the yyyy-mm-dd HH:MM format.

Additionally, as far as I can see, those settings only apply to the X-axis, not to the Y-axis.

**Comment by chrisjlocke** [comment]
If the cells are empty so DB4S hasn't guessed correctly, does it matter if its guessed incorrectly?  Once the cells have data, it'll then choose the right type?
